documentation/Data-View-Filters.md

```diff
@@ -0,0 +1,383 @@
+`[ fullName, email ]`
+
+Any text entered into the Search input will be matched to the values of these columns, using partial string matching (case insensitive):
+
+- search text "arl" would match "Carl", "Arlo", "earl@myplace.com"
+
+These searches *only* use text matching though, so the specified search columns should contain text data. If you need to narrow the results by more complex data structures, then you can specify [Filter definitions](#filters)
+
+If no `table_search_columns` (i.e. `null`) are provided, the "Search" input will not appear in the front end.
+
+### API
+
+Search text is passed to the server as part of the query parameters of the `data-views/<table>` endpoint, e.g.
+
+```
+/data-views/product?search=arl
+```
+
+## Sorting
+
+In the front-end, clicking on a column header will sort the table by that column (and clicking again reverses the order). If the column is a "basic" column (i.e. maps to an actual database column), then the default sort should be adequate, and no specific definition is required. However, if you have a column whose data is defined by a [value_expression](Data-View.md#data_view_column_definition-table) then you will need to define how it should be sorted. (If not, then the user will receive a "Column not sortable" notification when they try to sort by it.)
+
+We do this by specifying *another* column in the field `sort_column` for `data_view_column_definition`, which is what the database will sort by when clicking on this column.
+
+For example, if you have an "Address" column, that is build by combining a number of different fields (e.g. `address`, `province`, `country`), then you could specify any one of these as the value to sort by when clicking your custom "Address" column.
+
+It's also possible to convert complex data types to simplified "text" (or other simple data) versions of the values (see [Handling complex data structures](#handling-complex-data-structures) for more info), in which case you can specify this simplified column as the sort column for the more complex display value.
+
+### API
+
+Sorting requests are passed to the server as part of the query parameters of the `data-views/<table>` endpoint, e.g.
+
+```
+/data-views/product?orderBy=name&ascending=false
+```
+
+## Filters
+
+In the front end, there are several diffent UI elements for filtering table data. Which one is shown for a given data value is determined by a combination of the data type and explicit filter definitions specified in `data_view_column_definition`.
+
+### Handling complex data structures
+
+One of the main challenges with trying to create filters is that much of the data stored in tables is not easily filterable. Many fields store the entire "response" object that came from an application question, which could be any complex shape. For example, a `genericNames` field that has captured responses from a "Search" element might have data stored like so:
+
```

Comment:
This is quite technical, and I'd planned to put it *after* the filters section, but it became too hard to talk about much of the filters detail without an explanation of what "filterData" columns are for.

Reply:
The descriptions are very good and succinct, so looks good for SystemAdmin users - who this Docs should be used by